### ISSUE TITLE
Inconsistency in temperatures used to calculate column densities in beam heating?

### ISSUE THREAD

**Issue by jwreep** [rice-solar-physics/HYDRAD#104]
The electron temperature is used to determine whether the a cell is below `OPTICALLY_THICK_TEMPERATURE` when calculating column densities:
https://github.com/rice-solar-physics/HYDRAD/blob/0fa1f9949540e6b9379548c72a030621fdee381d/HYDRAD/source/eqns.cpp#L2116-L2128

but the LTE chromosphere uses the hydrogen temperature, and even has a comment about using consistent rates:

https://github.com/rice-solar-physics/HYDRAD/blob/0fa1f9949540e6b9379548c72a030621fdee381d/HYDRAD/source/eqns.cpp#L1160-L1178

Perhaps one could argue that the temperatures should be quickly collisionally equilibrated, but I think we could sidestep any potential issues by changing this line to use the hydrogen temperature: 
https://github.com/rice-solar-physics/HYDRAD/blob/0fa1f9949540e6b9379548c72a030621fdee381d/HYDRAD/source/eqns.cpp#L2117

Or am I missing something?

**Comment by jwreep** [comment]
I'm guess I'm missing that the NLTE chromosphere seems to use the electron temperature to calculate excitation rates, so perhaps we might want to check all of this for consistency.

**Comment by sjbradshaw** [comment]
This is a difficult one because the electron density is needed in order to calculate the electron temperature from the electron energy. To do this, we first need the hydrogen population fractions, from which we can find the electron density (and add a small correction due to ionized trace elements in LTE), but of course the population fractions are a function of the electron temperature… So, there are two choices: 1. Use the proton/hydrogen temperature (assuming Ti ~ Te at high density), which we have ready access to; or 2. Use an initial guess for the electron temperature to find the hydrogen population fractions and then recalculate Te with the new electron density (or recalculate the electron energy with the new electron density, to keep Te the same).

 

I remember spending quite a while on this issue before settling on the way it is currently handled as a compromise. However, if it’s of concern then I recommend doing a few experiments along the lines of the above (including changing the switch between optically-thin / thick radiation to the proton/hydrogen temperature) to see what the differences are.

 

 

 

From: Jeffrey Reep ***@***.***> 
Sent: Thursday, March 14, 2024 10:12 PM
To: rice-solar-physics/HYDRAD ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [rice-solar-physics/HYDRAD] Inconsistency in temperatures used to calculate column densities in beam heating? (Issue #104)

 

The electron temperature is used to determine whether the a cell is below OPTICALLY_THICK_TEMPERATURE when calculating column densities:
https://github.com/rice-solar-physics/HYDRAD/blob/0fa1f9949540e6b9379548c72a030621fdee381d/HYDRAD/source/eqns.cpp#L2116-L2128

but the LTE chromosphere uses the hydrogen temperature, and even has a comment about using consistent rates:

https://github.com/rice-solar-physics/HYDRAD/blob/0fa1f9949540e6b9379548c72a030621fdee381d/HYDRAD/source/eqns.cpp#L1160-L1178

Perhaps one could argue that the temperatures should be quickly collisionally equilibrated, but I think we could sidestep any potential issues by changing this line to use the hydrogen temperature:
https://github.com/rice-solar-physics/HYDRAD/blob/0fa1f9949540e6b9379548c72a030621fdee381d/HYDRAD/source/eqns.cpp#L2117

Or am I missing something?

**Comment by jwreep** [comment]
I'll do some simple tests.  It might be completely unimportant, but it caught my attention this week.  I'm looking at implementing some return current heating, so reviewing what is currently in there.

**Comment by sjbradshaw** [comment]
Fair enough – at some point I meant to get around to implementing the chromospheric losses for flares too (Hong, Carlsson, Ding, 2022, A&A 661, A77).
